packages/ra-ui-materialui/src/button/BulkUpdateWithUndoButton.tsx

```diff
@@ -41,7 +41,6 @@ export const BulkUpdateWithUndoButton = (
                 undoable: true,
             });
             unselectAll();
-            refresh();
```

Comment:
The success refresh was necessary because the cache update was wrong. After the fix, the refresh is not only unnecessary, it also poses a problem in optimistic mode.

In this mode, the button updates the cache, then refreshes (and shows the old version), then proceeds to the mutation, which properly updates the value. This leads to a misleading blink:

old value => new value => old value => new value

We also remove the refresh in the BulkUpdateWithConfirmButton as it can be used in optimistic mode.